source/agora/test/EnrollmentManager.d

```diff
@@ -74,4 +75,12 @@ unittest
     // check if nodes contains enrollment data previously sent.
     nodes.each!(node =>
         retryFor(node.hasEnrollment(enroll.utxo_key) == true, 5.seconds));
+
+    // tests for revealing a pre-image
+    auto preimage = node_1.getPreimage(1000);
+    node_1.receivePreimage(preimage);
```

Comment:
This node should already know about its own preimages. It is odd that it would need to send the preimage to itself in order to know about it, no?